gprofiler/log.py

```diff
@@ -174,6 +182,12 @@ def try_send_log_to_server(self):
         # Snapshot the current num logs because logs list might be extended meanwhile.
         logs_count = len(self._logs)
         try:
+            if self._truncated:
+                self._logger.warning(
+                    f"Log buffer truncation has occured as the maximum number of records ({self.MAX_BUFFERED_RECORDS}) "
+                    "was reached",
+                )
+                self._truncated = False
```

Comment:
append the log to the `_logs` list only if `self._truncated == False` in `RemoteLogsHandler.emit`, and set `self._truncated = False` only after you successfully send the logs home. Currently, you will add tons of `Log buffer truncation has occured` messages, which override older messages which would be more interesting.